test/e2e/pkg/exec/exec.go

```diff
@@ -0,0 +1,40 @@
+package exec
+
+import (
+	"context"
+	"fmt"
+	"os"
+	osexec "os/exec"
+)
+
+// Command executes a shell command.
+func Command(ctx context.Context, args ...string) error {
+	_, err := CommandOutput(ctx, args...)
+	return err
+}
+
+// CommandOutput executes a shell command and return the command's output.
+func CommandOutput(ctx context.Context, args ...string) ([]byte, error) {
+	//nolint: gosec
+	// G204: Subprocess launched with a potential tainted input or cmd arguments
+	cmd := osexec.CommandContext(ctx, args[0], args[1:]...)
+	out, err := cmd.CombinedOutput()
+	switch err := err.(type) {
+	case nil:
```

Comment:
is it just me or do you also find it weird that you can test a type against nil?

Reply:
This was coming from William's patch, so I assumed it was good.
I've just taken a look at our code base, and we do this quite often, e.g. [here,](https://github.com/cometbft/cometbft/blob/bd9d7e/light/client.go#L731) [here](https://github.com/cometbft/cometbft/blob/bd9d7e/evidence/reactor.go#L88), or [here](https://github.com/cometbft/cometbft/blob/4790ea3e46475064d5475c787427ae926c5a9e94/light/detector.go#L59).

Reply:
Seems to be a standard way to handle errors...

Reply:
It seems to be one of Go's idiosyncrasies